Models/MuZero_torch_agent.py

```diff
@@ -221,24 +231,50 @@ class MultiMlp(torch.nn.Module):
     def __init__(self,
                  input_size,
                  layer_size,
-                 layer_num,
                  output_sizes,
                  output_activation=torch.nn.Identity,
-                 activation=torch.nn.ReLU):
+                 activation=torch.nn.LeakyReLU):
         super().__init__()
 
-        # One linear that encodes the observation
-        self.encoding_layer = torch.nn.Sequential(*[torch.nn.Linear(input_size, layer_size),
-                                                    activation()] * layer_num).cuda()
+        layers = []
+        layers += [torch.nn.Linear(input_size, layer_size[0]), activation()]
+        for i in range(len(layer_size) - 1):
+            layers += [torch.nn.Linear(layer_size[i], layer_size[i + 1]), activation()]
+        self.encoding_layer = torch.nn.Sequential(*layers).cuda()
 
-        self.output_heads = []
+        # self.output_heads = []
 
-        for size in output_sizes:
-            output_layer = torch.nn.Sequential(
-                torch.nn.Linear(layer_size, size),
+        self.head_0 = torch.nn.Sequential(
+                torch.nn.Linear(layer_size[-1], output_sizes[0]),
                 output_activation()
             ).cuda()
-            self.output_heads.append(output_layer)
+        
+        self.head_1 = torch.nn.Sequential(
```

Comment:
Is there any reason why using a loop wouldn't work here?

Reply:
For some reason, pytorch does not detect layers inside a list as part of the network (If you check the model's weights, there are not weights for the heads if put inside a list)

Reply:
I have a hard time believing this. Maybe we just initialized something a little incorrectly before? If that's the case, so be it I guess

Reply:
Yea you're right thanks for bringing that up. Looks like pytorch has something called `ModuleList` that we're supposed to use instead.

How about something like this then?

```python
class MultiMlp(torch.nn.Module):
    def __init__(self,
                 input_size,
                 layer_size,
                 layer_num,
                 output_sizes,
                 output_activation=torch.nn.Identity,
                 activation=torch.nn.ReLU):
        super().__init__()

        # One linear that encodes the observation
        self.encoding_layer = torch.nn.Sequential(*[torch.nn.Linear(input_size, layer_size),
                                                    activation()] * layer_num).cuda()

        self.output_heads = torch.nn.ModuleList([
            torch.nn.Sequential(
                torch.nn.Linear(layer_size, size),
                output_activation()
            ) for size in output_sizes
        ]).cuda()

    def forward(self, x):
        # Encode the hidden state
        x = self.encoding_layer(x)

        output = [
          head(x)
          for head in self.output_heads
        ]

        return output

    def __call__(self, x):
        return self.forward(x)
```